### ISSUE TITLE
How to load pretrained weights for Transformer Encoder & Finetune

### ISSUE THREAD

**Issue by albertobeto** [facebookresearch/ParlAI#4118]
Hi!
I trained a classifier from parlai/agents/transformer/classifier.py 
I include the training command at the bottom.

The encoder has only 4 layers, so I consider it a "small" model.
I was wondering, running the classifier would start the weights from scratch, right?
I assume my accuracy would be higher if I can load weights from another task (for example, the first 4 layers of BERT), freeze these weights and train only the classifying head. Then, finetune the whole model.

Checking the training flags, I see there are some interesting ones:
https://parl.ai/docs/agent_refs/transformer.html
--update-classifier-head-only
--load-from-pretrained-ranker
--init-model  # Initialize model with weights from this file.

I could train in two steps, using the first flag to fix the encoder weights.
However, I'm not sure how to extract the lower layers from other model (BERT), or use the encoder from another model of similar size (Image Chat), and if load from ranker or init-model would be the better option for loading such weights.

Any help is much appreciated :D

Script for training the model:

parlai train_model \
--model transformer/classifier \
--model-file /location/in/my/computer \
--task my_task \
--classes_from_file "/path/to/classes/file.txt"  \
--batchsize 128  \
--embedding_size 600  \
--ffn_size 4096  \
--n_layers 4  \
--n_heads 6  \
--n_positions 128  \
--optimizer "adam"  \
--relu_dropout 0.0  \
--save_after_valid true  \
--text_truncate 128  \
--truncate 128  \
--variant "prelayernorm"  \
--learningrate 5e-06  \
...
